File: Day7/StrongNumber.java
import java.util.Scanner;
public class StrongNumber {
    
    // function to check if a number is strong  

    public static int factorial(int n){

        int factorial=1;
        while(n != 0){
            factorial *= n;
            n--;
        }
        return factorial;
    }
    // function to check if a number is strong 

    public static int getIsStrongNumber(int num){

        if(num <0){           
            return -1;
        }
        if(num ==  0){
            return 0;
        }
        int temp = num;
        int sum =0;
        while(num != 0){
            int digit = num % 10;
            sum += factorial(digit);
            num /= 10;
        }
        if(temp == sum){
            return 1;
        }
        return 0;
    }
    public static void main(String[] args) {
    
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the Number: ");
        int num = sc.nextInt();
        sc.close();
        int ans = getIsStrongNumber(num);

        if (ans == 1 ) {
            System.out.println(num + "  is a Strong Number.");
        }
        else if (num == 0){
            System.out.println(num + "  is not a Strong Number.");
        }else{
        System.out.println("strong number are not defined for negative number");
        }
    }
}

File: Day7/strongNumber.js
// function to calculate factorial 

function calculateFactorial(n){
    let fact =1;
    while (n > 0){
        fact *= n;
        n--;
    }
    return fact;
}
// function to check if a number is strong 

function isStrongNumber(num){

    if(num <0){      
        console.log("Strong number are not defined for negative number");
        return false;
    }
    if(num ==  0){
        return false;
    }
    let temp = num ,  sum = 0;
    while( num > 0){
        sum += calculateFactorial(num % 10);
        num = Math.floor(num/10);
    }
    return temp == sum;
}
let num = parseInt(prompt("Enter the Number: "));
let res= isStrongNumber(num);
if (res)
    console.log(num + " is a Strong Number.");
else
    console.log(num + " is a not Strong Number.");
